src/callTraceStorage.cpp

```diff
@@ -167,12 +167,12 @@ void CallTraceStorage::collectSamples(std::map<u64, CallTraceSample>& map) {
 }
 
 // Adaptation of MurmurHash64A by Austin Appleby
-u64 CallTraceStorage::calcHash(int num_frames, ASGCT_CallFrame* frames) {
+u64 CallTraceStorage::calcHash(int num_frames, ASGCT_CallFrame* frames, bool truncated) {
     const u64 M = 0xc6a4a7935bd1e995ULL;
     const int R = 47;
 
     int len = num_frames * sizeof(ASGCT_CallFrame);
-    u64 h = len * M;
+    u64 h = len * M * (truncated ? 1 : 2);
```

Comment:
I think `+` makes more sense. Otherwise, a truncated stacktrace will have the same initial value as a full stack trace two times shorter. As an edge case, it will not be possible to distinguish truncated and non-truncated zero-length stack traces.

Reply:
Good point.
Would it then make sense to do something like `u64 h = 2 * len * M + (truncated ? 0 : 1)` so we have clear disambiguation between truncated and non-truncated?